tests/st/utils/docker_host.py

```diff
@@ -250,7 +253,24 @@ def start_calico_node(self, options=""):
         args.append(options)
 
         cmd = ' '.join(args)
-        self.calicoctl(cmd)
+
+        if with_ipv4pool_cidr_env_var:
+            # Run the dryrun command, then modify and execute the command that
+            # that tells us.
+            output = self.calicoctl(cmd)
+            base_command = output.split('\n')[-4].rstrip()
```

Comment:
This feels a little fragile to me.  I think we'd be better off making use of #1511 - which is almost good to go.

I've suggested in #1511 that the start_calico_node() is extended to take list of environments.  You could make the default list just contain the CALICO_IPV4POOL_CIDR environment thus most tests would run with the modified pool CIDR, but we'd have an easy way of overriding it if, for example, we were testing what the actual default value was.